src/renderers/dom/client/syntheticEvents/SyntheticEvent.js

```diff
@@ -156,13 +173,45 @@ assign(SyntheticEvent.prototype, {
    * `PooledClass` looks for `destructor` on each instance it releases.
    */
   destructor: function() {
+    var setNull = setToNullOrWarning.bind(this);
     var Interface = this.constructor.Interface;
-    for (var propName in Interface) {
-      this[propName] = null;
+    for (var interfacePropName in Interface) {
+      setNull(interfacePropName);
+    }
+    otherNullableProps.forEach(setNull);
+    this._destructored = true;
+
+    function setToNullOrWarning(propName) {
```

Comment:
Sounds good to me. So how about I put this function at the bottom of the file, then change the code to something like:

```javascript
var Interface = this.constructor.Interface;
for (var interfacePropName in Interface) {
  setToNullOrWarning(this, interfacePropName);
}
setToNullOrWarning(this, 'nativeEvent');
this.dispatchConfig = null;
this._targetInst = null;
this._destructored = true;
```

Where `setToNullOrWarning` accepts a `context` and a `propertyName`.